src/zombiereborn.cpp

```diff
@@ -515,24 +540,28 @@ void ZR_CheckWinConditions(bool bCheckCT)
 	}
 
 	// didn't return early, all players on one or both teams are dead.
-	g_pGameRules->TerminateRound(5.0f, bCheckCT ? CSRoundEndReason::TerroristWin : CSRoundEndReason::CTWin);
+	// allow the opposite team to win
+	iTeamNum = iTeamNum == CS_TEAM_CT ? CS_TEAM_T : CS_TEAM_CT;
+	ZR_EndRoundAndAddTeamScore(iTeamNum);
+}
+
+void ZR_EndRoundAndAddTeamScore(int iTeamNum)
+{	
+	CSRoundEndReason iReason = CSRoundEndReason::Draw;
+	if (iTeamNum == CS_TEAM_T)
+		iReason = CSRoundEndReason::TerroristWin;
+	else if (iTeamNum == CS_TEAM_CT)
+		iReason = CSRoundEndReason::CTWin;
+	// CONVAR_TODO
+	// use mp_round_restart_delay here
+	g_pGameRules->TerminateRound(5.0f, iReason);
 	g_ZRRoundState = EZRRoundState::ROUND_END;
 	ToggleRespawn(true, false);
 
-	// increment team score
-	iTeamNum = bCheckCT ? CS_TEAM_T : CS_TEAM_CT; //?????
-
-	// so uhh.. I don't know how to get cteam the proper way..
-	CTeam* pTeam = nullptr;
-	while (nullptr != (pTeam = (CTeam*)UTIL_FindEntityByClassname(pTeam, "cs_team_manager")))
-	{
-		if (pTeam->m_iTeamNum() == iTeamNum)
-		{
-			pTeam->m_iScore = pTeam->m_iScore() + 1;
-			//Msg("%d, %d\n", pTeam->m_iTeamNum(), pTeam->m_iScore());
-			return;
-		}
-	}
+	if (iTeamNum == CS_TEAM_CT)
+		g_hTeamCT->m_iScore = g_hTeamCT->m_iScore() + 1;
+	else if (iTeamNum == CS_TEAM_T)
+		g_hTeamT->m_iScore = g_hTeamT->m_iScore() + 1;
```

Comment:
You should probably still verify these handles are valid, and log an error/return if not. To save us from debugging crashes later on if we somehow fail to find `cs_team_manager` entities at map start.